### DISCUSSION TITLE
.weeks() gives 0 for months with daylight saving time time change

### DISCUSSION THREAD

**Discussion by themtok** [moment/moment#6417] [Q&A] [answered]
**Describe the bug**
For PL locale, for month Nov, duration made with month start and end weeks() method always returns 0

**Expected behavior**
weeks() should return 4 or 3

**Desktop (please complete the following information):**
 - Windows
 - Browser: Edge
 - Version 2.29.4

**Moment-specific environment**

* Warsaw
* 26/01/2023 3pm

Please run the following code in your environment and include the output:
```
s=moment('2023-10-01')
e = moment('2023-10-01').endOf("month");
dur = moment.duration(e.diff(s))
dur.weeks()
```

**Comment by daniellecrasiuc** [comment]
Hi can I be assigned to this Issue?

**Comment by daniellecrasiuc** [comment]
Hi, I have been trying to replicate this issue but am getting the expected behavior, is the issue still persisting when you run it?

**Comment by mattjohnsonpint** [comment, marked as answer]
Thanks for the example! `.weeks()` returns the residual week component of a normalized duration, not total elapsed weeks. I'd use `duration.asWeeks()` for the total, optionally wrapped in `Math.floor`. The posted Warsaw interval is about `4.43` weeks; DST accounts for the extra elapsed hour.
